demos/inpage/src/accounts/SafeECDSAAccountWrapper.ts

```diff
@@ -135,4 +148,114 @@ export default class SafeECDSAAccountWrapper implements IAccount {
+
+    const safeProxy = Safe__factory.connect(this.address, owner);
+    const safeProxyAddress = await safeProxy.getAddress();
+
+    const viewer = new SafeSingletonFactoryViewer(owner, chainId);
+    const recoveryPluginDeployed = await viewer.isDeployed(
+      SafeECDSARecoveryPlugin__factory,
+      [safeProxyAddress, recoveryAddress],
+    );
+
+    let recoveryPlugin: SafeECDSARecoveryPlugin;
+    if (recoveryPluginDeployed) {
+      recoveryPlugin = viewer.connectAssume(SafeECDSARecoveryPlugin__factory, [
+        safeProxyAddress,
+        recoveryAddress,
+      ]);
+    } else {
+      const admin = await this.waxInPage.requestAdminAccount(
+        'deploy-contracts',
+      );
+      const factory = await SafeSingletonFactory.init(admin);
+
+      // TODO: (merge-ok) deploy recovery plugin on initial contracts deploy
+      recoveryPlugin = await factory.connectOrDeploy(
+        SafeECDSARecoveryPlugin__factory,
+        [safeProxyAddress, recoveryAddress],
+      );
+    }
+
+    const recoveryPluginAddress = await recoveryPlugin.getAddress();
+
+    let moduleEnabled = await safeProxy.isModuleEnabled(recoveryPluginAddress);
+    if (!moduleEnabled) {
+      // TODO: (merge-ok) try calling 'enableModule' via bundler
+      await executeContractCallWithSigners(
+        safeProxy,
+        safeProxy,
+        'enableModule',
+        [recoveryPluginAddress],
+        [owner],
+      );
+
+      moduleEnabled = await safeProxy.isModuleEnabled(recoveryPluginAddress);
+      assert(moduleEnabled, 'module not enabled');
+
+      this.recoveryAddress = recoveryAddress;
+      await this.waxInPage.storage.accounts.set([this.toData()]);
+    }
+  }
+
+  async recoveryAccount(
+    newOwnerAccountSeedPhrase: string,
+    recoveryAddressSeedPhrase: string,
+  ) {
+    const provider = this.waxInPage.ethersProvider;
+    const owner = new ethers.Wallet(this.privateKey, provider);
+    const { chainId } = provider._network;
+
+    const newOwnerWallet = ethers.Wallet.fromPhrase(
+      newOwnerAccountSeedPhrase,
+      provider,
+    );
+    const newOwnerAddress = newOwnerWallet.address;
+
+    const recoveryWallet = ethers.Wallet.fromPhrase(
+      recoveryAddressSeedPhrase,
+      provider,
+    );
+
+    const safeProxy = Safe__factory.connect(this.address, owner);
+    const safeProxyAddress = await safeProxy.getAddress();
+
+    const viewer = new SafeSingletonFactoryViewer(owner, chainId);
+    const recoveryPlugin = await viewer.connectOrThrow(
+      SafeECDSARecoveryPlugin__factory,
+      [safeProxyAddress, recoveryWallet.address],
+    );
+
+    const plugin = this.getContract();
+
+    // TODO: (merge-ok) ensure wallet is already funded
+    await (
+      await owner.sendTransaction({
+        to: recoveryWallet.address,
+        value: ethers.parseEther('0.1'),
+      })
+    ).wait();
+
+    const pluginAddress = await plugin.myAddress();
+    await recoveryPlugin
+      .connect(recoveryWallet)
```

Comment:
We need the recovery account seed phrase or private key as we need to execute `resetEcdsaAddress` as the recovery account